### PR TITLE
Extend jmalloc to monitor the peak memory usage of each thread (11.2.x)

### PR DESCRIPTION
**Pull request by fwyzard**
See https://github.com/jemalloc/jemalloc/pull/1176 and https://github.com/jemalloc/jemalloc/issues/1850 for the discussion with the jemalloc development team.

### CODE REVIEW

**Comment by fwyzard** [comment]
@cmsbuild, please test

File: jemalloc.spec [hunk 1]
```diff
@@ -1,8 +1,8 @@
-### RPM external jemalloc 5.2.1
+### RPM external jemalloc 5.2.1-peakmem
 
 %define tag %{realversion}
-%define branch master
-%define github_user jemalloc
+%define branch 52x-peakmem
+%define github_user fwyzard
```
Comment (makortel) [inline jemalloc.spec]: If we go with the patch, we should probably fork `jemalloc` under https://github.com/cms-externals/

Comment (fwyzard) [reply jemalloc.spec]: Sure - this is just a way to show what the changes would be, and make sure it doesn't break anything in CMSSW.

**Comment by smuzaffar** [comment]
please test

**Comment by smuzaffar** [comment]
+externals

**Pull request merged**